### ISSUE TITLE
Possible backward compatibility issue ?

### ISSUE THREAD

**Issue by nouryf** [tobinbradley/mbtiles-server#8]
I created an .mbtiles using Openmaptiles tools (max zoom level: 14) and was serving it's tiles using 'mbtiles-server' (version dated July 2020). All is working fine.

Recently I started using the new version of mbtiles-server (current `master`  with last commits until May 31, 2022).

I noticed that with the old version that tiles at zoom level 15 and + are rendered fine (over zooming I assume since they don't really exist at the .mbtiles file). However with the new `mbtiles-server` those tiles are not rendered and I get code 204.

Old version: [https://morocco-maps.com/old.html](https://morocco-maps.com/old.html)
New version: [https://morocco-maps.com/new.html](https://morocco-maps.com/new.html)

The map is a demo, so the font are not there, but with the hash code, you can see the difference in behavior:
[https://morocco-maps.com/old.html#16.67/33.57565/-7.60155](https://morocco-maps.com/old.html#16.67/33.57565/-7.60155)
[https://morocco-maps.com/new.html#16.72/33.58488/-7.58348](https://morocco-maps.com/new.html#16.72/33.58488/-7.58348)

Is there is way with the new `mbtiles-server` version to get back the old behavior and let the app overzoom locally instead of having code 204 ?

Thanks

**Comment by tobinbradley** [comment]
Try adding a [maxzoom to your tile source](https://docs.mapbox.com/mapbox-gl-js/style-spec/sources/), it should fix the problem:

```json
"sources": {
    "morocco": {
      "type": "vector",
      "tiles": ["https://morocco-maps.com/App/morocco/{z}/{x}/{y}.pbf"],
      "maxzoom": 14
    }
  }
```

It'll have the added bonus of not having a bunch (zoom levels 15-22) of requests for tiles you don't have hitting your server when people zoom in.

tl;dr the server doesn't do any tile over-zooming, that's handled on the client. My guess is the old version of mbtiles-server returned a 404 (not found), whereas the latest version returns a 204 (no data), and the client is handling those http codes differently. You could change 204 back to a 404 in `index.js`, but the best practice is to set your tile source min and max zoom levels whenever your tile min is more than 0 and your tile max is less than 22. It'll still draw the layers past your max zoom, it just won't make requests for tiles you don't have, which will both solve the problem and get rid of unnecessary server requests.

**Comment by nouryf** [comment]
Thanks @tobinbradley for the quick and the detailed answer. Adding "maxzoom": 14 to the .json file did the trick. As you explained it looks like it's the best solution for this issue. Thanks again.
